### PR TITLE
Mark log variables as core and add documentation

### PR DESCRIPTION
**Pull request by krichardsson**
Document log variables and select a set of variable to be considered core logs. Core logs will be part of the stable API.

### CODE REVIEW

File: src/modules/src/estimator_kalman.c [hunk 1]
```diff
@@ -447,34 +447,121 @@ void estimatorKalmanGetEstimatedRot(float * rotationMatrix) {
   memcpy(rotationMatrix, coreData.R, 9*sizeof(float));
 }
 
-// Stock log groups
+/**
+ * Variables and results from the Extended Kalman Filter
```
Comment (jonasdn) [inline src/modules/src/estimator_kalman.c:451]: maybe link here? https://www.bitcraze.io/documentation/repository/crazyflie-firmware/2020.04/functional-areas/state_estimators/

File: src/modules/src/extrx.c [hunk 2]
```diff
@@ -177,14 +177,42 @@ static void extRxDecodeSpektrum(void)
 
 /* Loggable variables */
 #ifdef ENABLE_EXTRX_LOG
+/**
+ * External receiver (RX) log group. This contains received raw
+ * channel data as well as RPTY data after it has been converted.
```
Comment (jonasdn) [inline src/modules/src/extrx.c:182]: What is RPTY?

File: src/hal/src/pm_stm32f4.c [hunk 3]
```diff
@@ -407,16 +407,46 @@ void pmTask(void *param)
...
+/**
+ * @brief BigQuad external current measurement [V]
+ */
 LOG_ADD(LOG_FLOAT, extCurr, &extBatteryCurrent)
+/**
+ * @brief Battery charge current [A]
+ */
 LOG_ADD(LOG_FLOAT, chargeCurrent, &pmSyslinkInfo.chargeCurrent)
+/**
+ * @brief PM state [battery=0, charging=1, charged=2, lowPower=3, shutDown=4]
```
Comment (jonasdn) [inline src/hal/src/pm_stm32f4.c:439]: In other places we use a table for documentation, something like this?

```
@brief State of power management

State | Meaning
----------------------
0     | battery
1     | charging
2     | charged
3     | low power
4     | shutdown
```

**Review by jonasdn — CHANGES_REQUESTED** [review]

File: src/modules/src/stabilizer.c [hunk 4]
```diff
@@ -433,45 +559,165 @@ LOG_GROUP_START(controller)
...
 
+/**
+ * Log group for the state estimator, the currently estimated state of the platform.
+ *
+ * Note: all values may not be updated depending on which estimator that is used.
+ */
 LOG_GROUP_START(stateEstimate)
+
+/**
+ * @brief The position of the Crazyflie in the global reference frame, X [m]
```
[the remark this answers is not in the document]
Comment (krichardsson) [reply src/modules/src/stabilizer.c:581]: Good question! The bolt is called the Crazyflie Bolt so from that point of view it is OK I think.

[the remark this answers is not in the document]
Comment (krichardsson) [reply src/modules/src/stabilizer.c]: Used platform instead

File: src/modules/src/controller_pid.c [hunk 5]
```diff
@@ -152,25 +152,80 @@ void controllerPid(control_t *control, setpoint_t *setpoint,
...
 }
 
-
+/**
+ * Logging variables for the command and reference signals for the 
+ * altitude PID controller
+ */
 LOG_GROUP_START(controller)
+/**
+ * @brief Trust command
```
[the remark this answers is not in the document]
Comment (krichardsson) [reply src/modules/src/controller_pid.c]: fixed

File: src/modules/src/tdoaEngineInstance.c [hunk 6]
```diff
@@ -32,17 +32,62 @@
...
+ */
 STATS_CNT_RATE_LOG_ADD(stCc, &tdoaEngineState.stats.clockCorrectionCount)
+
+/**
+ * @brief Rate of hits when looking up anchor contexts for packets [1/s]
+ */
 STATS_CNT_RATE_LOG_ADD(stHit, &tdoaEngineState.stats.contextHitCount)
+
+/**
+ * @brief Rate of misses when looking up anchor contexts for packets [1/s]. If this number is high, the CF is receiving packets from more anchors than can be stored in the TDoA storage.
```
[the remark this answers is not in the document]
Comment (krichardsson) [reply src/modules/src/tdoaEngineInstance.c]: Fixed

File: src/modules/src/tdoaEngineInstance.c [hunk 7]
```diff
@@ -32,17 +32,62 @@
...
+ */
 LOG_ADD(LOG_FLOAT, cc, &tdoaEngineState.stats.clockCorrection)
+
+/**
+ * @brief The Time Of Flight from anchor A to anchor B (including antenna delay), as measured by anchor A [UWB radio ticks]. A is selected using the tdoaEngine.logId parameter and B is selected by tdoaEngine.logOthrId.
+ */
 LOG_ADD(LOG_UINT16, tof, &tdoaEngineState.stats.tof)
+
+/**
+ * @brief The difference in distance to anchor A and B, as measured by the Crazyflie [m]. A is selected using the tdoaEngine.logId parameter and B is selected by tdoaEngine.logOthrId.
```
[the remark this answers is not in the document]
Comment (krichardsson) [reply src/modules/src/tdoaEngineInstance.c]: Fixed

File: src/modules/src/estimator_kalman.c (hunk 1 above)
Comment (krichardsson) [reply src/modules/src/estimator_kalman.c:451]: Good question if we want (how many) links back to the web docs. They may be hard to maintain over time when we update the web? 
If there is a link it should link to master though: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/state_estimators/

File: src/hal/src/pm_stm32f4.c (hunk 3 above)
Comment (krichardsson) [reply src/hal/src/pm_stm32f4.c]: Fixed

File: src/modules/src/extrx.c (hunk 2 above)
Comment (knmcguire) [reply src/modules/src/extrx.c:182]: roll pitch trust yaw I guess? It's for attitude control

Comment (krichardsson) [reply src/modules/src/extrx.c:182]: Roll Pitch Yaw Thrust is my guess

**Review by jonasdn — APPROVED** [review]

**Pull request merged**